websurvey/src/routes/Form.svelte

```diff
@@ -0,0 +1,51 @@
+<script>
+    import { createEventDispatcher } from "svelte";
+    import { navigate } from "svelte-routing";
+
+    import MultipleChoice from "../components/MultipleChoice.svelte";
+    import ShortText from "../components/ShortText.svelte";
+    import typeformData from "../typeformData.js";
+
+    export let currentIndex;
+    export let ref;
+
+    let dispatch = createEventDispatcher();
+
+    const { fields } = typeformData;
+    const { length } = fields;
+
+    const handleSubmit = () => {
+        if (currentIndex < fields.length - 1) {
+            currentIndex += 1;
+            navigate(`/${ref}`, { replace: true });
+        }
+        currentIndex;
+        dispatch("indexUpdate", currentIndex);
+    };
+</script>
+
+<div class="surveyapp stack-large">
+    <form on:submit|preventDefault={handleSubmit}>
+        <div class="stack-small">
+            <!-- Question -->
+            {#each fields as field, index (field.id)}
+                {#if currentIndex === index}
```

Comment:
Ideally, given you have the currentIndex, you should be able to make `field` a variable that "reacts" to the change in currentIndex and is simply `field = fields[currentIndex]`, no? Rather than loop over it all?

Reply:
Agree that makes sense. Although I can't quite get this to run as it should. 

Currently, we have `field = fields[index]` with a prepended `:$` which Svelte interprets to mean 're-run this code whenever any of the referenced values change', which in our case is the `index`.

When you `$: console.log(field)` at the global scope in the `Form` component, the correct data object is printed on form submit. 

Similarly when you `$: console.log(field)` on the `ShortText` and `MultipleChoice` components, the correct data object is also printed.

However, the issue here is that the DOM isn't reacting to the change in state when two or more fields of the same `type` are next to each other in the array – I tested this by playing around with the JSON and adding in some more fields. 

On refresh, the data for the current object does render correctly which makes me think this could be do to with the way I've set up the `handleSubmit`.

The only workaround that I've found for this is to leave in the `{#each}` loop and compare each item with the `field` in an `{#if}` statement.

Will take this out though once the solution eventually comes to me!